File: backend/GeoQuiz_backend/GeoQuiz_backend/Domain/Entities/Achievement.cs
using GeoQuiz_backend.Domain.Entities;
using System.ComponentModel.DataAnnotations;

namespace GeoQuiz_backend.Domain.Entities
{
    public class Achievement
    {
        public Guid Id { get; set; }
        public string Code { get; set; } = null!;
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string Icon { get; set; } = null!;
        public AchievementCategory Category { get; set; }
        public AchievementRarity Rarity { get; set; }
        public ICollection<UserAchievement> UserAchievements { get; set; } = new List<UserAchievement>();
    }
    public enum AchievementCategory
    {
        General = 1,
        Gameplay = 2,
        Knowledge = 3,
        Streaks = 4,
        PvP = 5,
        Koth = 6,
        Special = 7
    }

    public enum AchievementRarity
    {
        Common = 1,      
        Rare = 2,        
        Epic = 3,        
        Legendary = 4    
    }
}

File: frontend/app/src/main/java/com/example/geoquiz_frontend/domain/entities/Achievement.java
package com.example.geoquiz_frontend.domain.entities;

import com.example.geoquiz_frontend.domain.enums.LocalizedText;

public class Achievement {
    public String code;
    public LocalizedText title;
    public LocalizedText description;
    public int category;
    public int rarity;
    public int progress;
    public boolean isUnlocked;
    public String unlockedAt;
    public String icon;

    public Achievement(String code, LocalizedText title, LocalizedText description, int category, int rarity,
                       int progress, boolean isUnlocked, String unlockedAt, String icon) {
        this.code = code;
        this.title = title;
        this.description = description;
        this.category = category;
        this.rarity = rarity;
        this.progress = progress;
        this.isUnlocked = isUnlocked;
        this.unlockedAt = unlockedAt;
        this.icon = icon;
    }
}
